python/tank_vendor/shotgun_base/utils.py

```diff
@@ -18,6 +18,34 @@
 from .log import get_shotgun_base_logger
 log = get_shotgun_base_logger()
 
+
+def with_cleared_umask(func):
+    """
+    Decorator which clears the umask for a method.
+
+    The umask controls default permissions for files
+    and folders. By setting the umask to zero, we can control
+    permissions.
+
+    Example use:
+
+    @with_cleared_umask
+    def my_io_method(param1, param2):
+
+        # do i/o operations in here
+    """
+    def inner(*args, **kwargs):
+        # set umask to zero, store old umask
+        old_umask = os.umask(0)
+        try:
+            # execute method payload
+            return func(*args, **kwargs)
+        finally:
+            # set mask back to previous value
+            os.umask(old_umask)
+    return inner
```

Comment:
I found [this](https://docs.python.org/2/library/functools.html#functools.wraps) recently. I recommend we use it when decorating methods. It will make our lives easier when we generate documentation from the source.